src/engine/RenderSystem.cpp

```diff
@@ -0,0 +1,145 @@
+#include "engine/RenderSystem.hpp"
+
+#include "util/GLDebug.hpp"
+#include <array>
+
+namespace lei3d {
+
+void RenderSystem::initialize(int width, int height) {
+    scwidth = width;
+    scheight = height;
+
+    forwardShader = Shader("./data/shaders/forward.vert", "./data/shaders/forward.frag");
+    postprocessShader = Shader("./data/shaders/screenspace_quad.vert", "./data/shaders/postprocess.frag");
+
+    glGenVertexArrays(1, &dummyVAO);
+
+    glGenFramebuffers(1, &FBO);
+    glBindFramebuffer(GL_DRAW_FRAMEBUFFER, FBO);
```

Comment:
Never saw the use of a frame buffer object, what a great example for it though. Learned a lot from this PR